### PR TITLE
Improve GCS signUrlOption javadoc

### PR DESCRIPTION
**Pull request by danielcompton**
This isn't quite ready for merge yet. I'm not sure if there is a better wording for the `withContentType` doc, and whether the `httpMethod` doc should actually go against `Storage.signUrl`?

### CODE REVIEW

**Comment by garrettjonesgoogle** [comment]
@danielcompton , did you want to finish this PR?

**Comment by danielcompton** [comment]
I think this is ready to merge if you're happy with the wording?

**Comment by garrettjonesgoogle** [comment]
@andreamlin can you verify the correctness of the new comments and merge if it looks good?

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java [hunk 1]
```diff
@@ -806,7 +806,7 @@ Object getValue() {
     }
 
     /**
-     * The HTTP method to be used with the signed URL.
+     * The HTTP method to be used with the signed URL. Defaults to GET.
```
Comment (andreamlin) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java]: Does it make sense to say "Defaults to GET" for a setter method? In what case would this method be called, but the default GET is used?

Comment (danielcompton) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java]: I'm trying to communicate that GET is the default method if this isn't called, not quite sure what wording I should use here? Maybe this?

> The HTTP method to be used with the signed URL. If this is not set, defaults to GET.

Comment (andreamlin) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java]: Sounds good!

Comment (danielcompton) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java]: Done, ready for re-review I think?

**Pull request merged**